File: examples/python/likee.py
import requests

API_BASE = 'https://cog.api.br/api/v1'

def download_likee(url: str) -> dict:
    """
    Download video from Likee
    
    Args:
        url: Likee video URL
        
    Returns:
        dict: Download data or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/likee/download', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            print('Likee Video Downloaded Successfully!')
            print(f"Title: {data['data']['title']}")
            print(f"Author: {data['data']['author']}")
            print(f"Views: {data['data']['views']}")
            print(f"Likes: {data['data']['likes']}")
            print(f"Comments: {data['data']['comments']}")
            print(f"Download URL: {data['data']['downloadUrl']}")
            print(f"Duration: {data['data']['duration']} seconds")
            print(f"Resolution: {data['data']['width']}x{data['data']['height']}")
            
            return data['data']
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

def get_likee_info(url: str) -> dict:
    """
    Get information about Likee video
    
    Args:
        url: Likee video URL
        
    Returns:
        dict: Video information or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/likee/info', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            print('Likee Video Information:')
            print(f"Title: {data['info']['title']}")
            print(f"Author: {data['info']['author']}")
            print(f"Views: {data['info']['views']}")
            print(f"Likes: {data['info']['likes']}")
            print(f"Comments: {data['info']['comments']}")
            print(f"Duration: {data['info']['duration']} seconds")
            print(f"Description: {data['info']['description']}")
            print(f"Resolution: {data['info']['width']}x{data['info']['height']}")
            
            return data['info']
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

if __name__ == '__main__':
    print('=== Likee Download Example ===\n')
    
    # Example 1: Download Likee video
    likee_url = 'https://likee.video/@username/video/123456'
    print(f'Downloading from Likee: {likee_url}')
    download_likee(likee_url)
    
    print('\n=== Likee Info Example ===\n')
    
    # Example 2: Get video info
    print(f'Getting video info: {likee_url}')
    get_likee_info(likee_url)

File: examples/nodejs/likee.js
const axios = require('axios');

const API_BASE = 'https://cog.api.br/api/v1';

/**
 * Download video from Likee
 * @param {string} url - Likee video URL
 */
async function downloadLikee(url) {
  try {
    const response = await axios.get(`${API_BASE}/likee/download`, {
      params: { url }
    });

    if (response.data.success) {
      console.log('Likee Video Downloaded Successfully!');
      console.log('Title:', response.data.data.title);
      console.log('Author:', response.data.data.author);
      console.log('Views:', response.data.data.views);
      console.log('Likes:', response.data.data.likes);
      console.log('Comments:', response.data.data.comments);
      console.log('Download URL:', response.data.data.downloadUrl);
      console.log('Duration:', response.data.data.duration, 'seconds');
      console.log('Resolution:', `${response.data.data.width}x${response.data.data.height}`);
      
      return response.data.data;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

/**
 * Get information about Likee video
 * @param {string} url - Likee video URL
 */
async function getLikeeInfo(url) {
  try {
    const response = await axios.get(`${API_BASE}/likee/info`, {
      params: { url }
    });

    if (response.data.success) {
      console.log('Likee Video Information:');
      console.log('Title:', response.data.info.title);
      console.log('Author:', response.data.info.author);
      console.log('Views:', response.data.info.views);
      console.log('Likes:', response.data.info.likes);
      console.log('Comments:', response.data.info.comments);
      console.log('Duration:', response.data.info.duration, 'seconds');
      console.log('Description:', response.data.info.description);
      console.log('Resolution:', `${response.data.info.width}x${response.data.info.height}`);
      
      return response.data.info;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

// Example usage
(async () => {
  console.log('=== Likee Download Example ===\n');
  
  // Example 1: Download Likee video
  const likeeUrl = 'https://likee.video/@username/video/123456';
  console.log('Downloading from Likee:', likeeUrl);
  await downloadLikee(likeeUrl);
  
  console.log('\n=== Likee Info Example ===\n');
  
  // Example 2: Get video info
  console.log('Getting video info:', likeeUrl);
  await getLikeeInfo(likeeUrl);
})();

module.exports = {
  downloadLikee,
  getLikeeInfo
};
